packages/web3-eth-accounts/src/common/utils.ts

```diff
@@ -0,0 +1,619 @@
+ * @return (Buffer)
+ */
+const setLength = function (msg: Buffer, length: number, right: boolean) {
+	const buf = zeros(length);
+	if (right) {
+		if (msg.length < length) {
+			msg.copy(buf);
+			return buf;
+		}
+		return msg.slice(0, length);
+	}
+	if (msg.length < length) {
+		msg.copy(buf, length - msg.length);
+		return buf;
+	}
+	return msg.slice(-length);
+};
+
+/**
+ * Throws if input is not a buffer
+ * @param {Buffer} input value to check
+ */
+export const assertIsBuffer = function (input: Buffer): void {
+	if (!Buffer.isBuffer(input)) {
+		// eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+		const msg = `This method only supports Buffer but input was: ${input}`;
+		throw new Error(msg);
+	}
+};
+/**
+ * Left Pads a `Buffer` with leading zeros till it has `length` bytes.
+ * Or it truncates the beginning if it exceeds.
+ * @param msg the value to pad (Buffer)
+ * @param length the number of bytes the output should be
+ * @return (Buffer)
+ */
+export const setLengthLeft = function (msg: Buffer, length: number) {
+	assertIsBuffer(msg);
+	return setLength(msg, length, false);
+};
+
+/**
+ * Trims leading zeros from a `Buffer`, `String` or `Number[]`.
+ * @param a (Buffer|Array|String)
+ * @return (Buffer|Array|String)
+ */
+const stripZeros = function (a: any): Buffer | number[] | string {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+	let first = a[0];
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
+	while (a.length > 0 && first.toString() === '0') {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment, prefer-destructuring, @typescript-eslint/no-unsafe-call, no-param-reassign
+		a = a.slice(1);
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, prefer-destructuring, @typescript-eslint/no-unsafe-member-access
+		first = a[0];
+	}
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+	return a;
+};
+
+/**
+ * Trims leading zeros from a `Buffer`.
+ * @param a (Buffer)
+ * @return (Buffer)
+ */
+export const unpadBuffer = function (a: Buffer): Buffer {
+	assertIsBuffer(a);
+	return stripZeros(a) as Buffer;
+};
+
+/**
+ * Converts a {@link Uint8Array} or {@link NestedUint8Array} to {@link Buffer} or {@link NestedBufferArray}
+ */
+export function arrToBufArr(arr: Uint8Array): Buffer;
+export function arrToBufArr(arr: NestedUint8Array): NestedBufferArray;
+export function arrToBufArr(arr: Uint8Array | NestedUint8Array): Buffer | NestedBufferArray;
+export function arrToBufArr(arr: Uint8Array | NestedUint8Array): Buffer | NestedBufferArray {
+	if (!Array.isArray(arr)) {
+		return Buffer.from(arr);
+	}
+	return arr.map(a => arrToBufArr(a));
+}
+
+/**
+ * Converts a {@link bigint} to a `0x` prefixed hex string
+ */
+export const bigIntToHex = (num: bigint) => `0x${num.toString(16)}`;
```

Comment:
[toHex](https://github.com/web3/web3.js/blob/de5617ce4f0f50785e075c7638deeac1d6ed615d/packages/web3-utils/src/converters.ts#L313)

Reply:
if replace base test is not working. error:
``` Web3ValidatorError: Web3 validator found 1 error[s]:
    value "90062145607137483435911490614552348104830437995083346642150762831216482558573" at "/0" must pass "int" validation
```
so our `toHex` has extra validation